node/core/approval-voting/src/ops.rs

```diff
@@ -311,3 +311,92 @@ pub fn force_approve(
 
 	Ok(approved_hashes)
 }
+
+/// Revert to the block corresponding to the specified `hash`.
+/// The operation is not allowed for blocks older than the last finalized one.
+pub fn revert_to(
+	overlay: &mut OverlayedBackend<'_, impl Backend>,
+	hash: Hash,
+) -> SubsystemResult<()> {
+	let mut stored_range = overlay.load_stored_blocks()?.ok_or_else(|| {
+		SubsystemError::Context("no available blocks to infer revert point height".to_string())
```

Comment:
While this is not the first time this pattern is used in this subsystem, we want to avoid any stringly typed errors and use error declarations akin to smth like https://github.com/paritytech/polkadot/blob/dc248c0664dcf4452984e21fa84515db457741ba/node/core/dispute-coordinator/src/error.rs#L31

Reply:
I see. 

Anyway the difference between approval-voting subsystem and dispute-coordinator (or chain-selection) subsystem is that the former doesn't have (yet) its `Error` type defined.
Every function here returns this "god" `SubsystemResult` type (with its god `SubsystemError type).

I can introduce a new `Error` type for the dispute-coordinator, but then it should be propagated to all the other methods/functions to keep coherence.

Maybe is better to do this in a separate PR?

Reply:
Agreed, it might be a bit too much for one PR